dftimewolf/metawolf/utils.py

```diff
@@ -0,0 +1,361 @@
+
+    Args:
+      value (str): The string to infer the type from.
+
+    Returns:
+      Any: The type for the string.
+    """
+    if self.Str2Bool(value) in [False, True]:
+      return bool
+
+    if self.IsInt(value):
+      return int
+
+    if self.IsFloat(value):
+      return float
+
+    return str
+
+  def CastToType(self, value: str, value_type: Any) -> Any:
+    """Cast a string to the desired type.
+
+    Returns None if the string cannot be cast to the desired type.
+
+    Args:
+      value (str): The string to cast.
+      value_type (Any): The type to cast the string to.
+    """
+    if value_type == int:
+      if not self.IsInt(value):
+        return None
+      return int(float(value))
+
+    if value_type == float:
+      if not self.IsFloat(value):
+        return None
+      return float(value)
+
+    if value_type == bool:
+      if self.Str2Bool(value) is None:
+        return None
+      return self.Str2Bool(value)
+
+    return value
+
+  @staticmethod
+  def Marshal(st: session.SessionSettable) -> Dict[str, Any]:
+    """Marshal a SessionSettable object to a JSON dictionary.
+
+    Args:
+      st (SessionSettable): The settable to marshal.
+
+    Returns:
+      Dict[str, Any]: A JSON dictionary representation of the settable.
+    """
+    return {
+        'session_id': st.session_id,
+        'recipe': st.recipe,
+        'name': st.name,
+        'description': st.description,
+        'value': st.GetValue(),
+        'type': st.type.__name__,
+        'optional': st.IsOptional()
+    }
+
+  @staticmethod
+  def Unmarshal(st: Dict[str, Any]) -> session.SessionSettable:
+    """Unmarshal a JSON dictionary to a SessionSettable object.
+
+    Args:
+      st (Dict[str, Any]): The JSON dictionary representation of the settable.
+
+    Returns:
+      SessionSettable: A SessionSettable object.
+    """
+    s = session.SessionSettable(
+        st['session_id'],
+        st['recipe'],
+        st['name'],
+        st['description'],
+        locate(st['type']),
+        optional=st['optional']
+    )
+    s.SetValue(st['value'])
+    return s
+
+  def ReadSessionFromFile(self, unmarshal: bool = True) -> Dict[str, Any]:
+    """Read Metawolf's sessions from file.
+
+    Args:
+      unmarshal (bool): Boolean that indicates whether or not the returned
+          dictionary should contain unmarshalled objects.
+
+    Returns:
+      Dict[str, Dict[str, Dict[str, Union[str, SessionSettable]]]]: A
+          dictionary that maps session IDs and recipes to their corresponding
+          settable (or JSON str).
+    """
+    # pylint: disable=line-too-long
+    loaded_sessions = {}  # type: Dict[str, Dict[str, Dict[str, Union[str, session.SessionSettable]]]]
+    # pylint: enable=line-too-long
+    if not os.path.exists(os.path.expanduser(self.session_path)):
+      return loaded_sessions
+
+    with open(os.path.expanduser(self.session_path), 'r') as f:
+      if f.read(1):
+        f.seek(0)
```

Comment:
What is this check meant to do? If you're just trying to see whether the file has contents in it, I would just call `loads` and catch any exception that this might raise and return `loaded_sessions`

Reply:
Good point, done